cloudinit/config/tests/test_grub_dpkg.py

```diff
@@ -0,0 +1,66 @@
+# This file is part of cloud-init. See LICENSE file for license information.
+
+import pytest
+
+from unittest import mock
+from cloudinit.util import ProcessExecutionError
+from cloudinit.config.cc_grub_dpkg import fetch_idevs
+
+
+class TestFetchIdevs():
+    """Tests cc_grub_dpkg.fetch_idevs()"""
+
+    # Note: udevadm info returns devices in a large single line string
+    @pytest.mark.parametrize(
+        "grub_output,udevadm_output,expected_idevs,expected_idevs_empty",
+        [
+            # Inside a container
+            (
+                ProcessExecutionError(),
+                '',
+                '',
+                'true',
+            ),
+            # KVM Instance
+            (
+                '/dev/vda',
+                '/dev/disk/by-path/pci-0000:00:00.0 /dev/disk/by-path/virtio-pci-0000:00:00.0',  # nopep8
```

Comment:
As we're already within parentheses, you can wrap these long lines like this:

```suggestion
                '/dev/disk/by-path/pci-0000:00:00.0'
                ' /dev/disk/by-path/virtio-pci-0000:00:00.0',
```

Alternatively, it might be easier to read if you made this a tuple (and then did `" ".join(udevadm_output)` in the body of the test):

```suggestion
                ('/dev/disk/by-path/pci-0000:00:00.0',
                 '/dev/disk/by-path/virtio-pci-0000:00:00.0')
```

Reply:
Thanks. Implemented in latest revision.